### PR TITLE
[REF] Add in upgrade step to populate missing contact_type.name field

### PR DESCRIPTION
**Pull request by seamuslee001**
Overview
----------------------------------------
As picked up by @aydun there was no fixing up of hte current name column data when it was missing as requested by @totten in https://github.com/civicrm/civicrm-core/pull/17548/files

Before
----------------------------------------
No populating of the name column if empty

After
----------------------------------------
Upgrade script with test

ping @eileenmcnaughton

### CODE REVIEW

File: CRM/Upgrade/Incremental/php/FiveTwentyEight.php [hunk 1]
```diff
@@ -69,4 +69,30 @@ public function setPostUpgradeMessage(&$postUpgradeMessage, $rev) {
...
+  public function upgrade_5_28_alpha1($rev) {
+    $this->addTask(ts('Upgrade DB to %1: SQL', [1 => $rev]), 'runSql', $rev);
+    $this->addTask('Populate missing Contact Type name fields', 'populateMissingContactTypeName');
+  }
+
+  public static function populateMissingContactTypeName() {
+    $contactTypes = \Civi\Api4\ContactType::get()
+      ->setCheckPermissions(FALSE)
+      ->execute();
+    foreach ($contactTypes as $contactTypes) {
```
Comment (aydun) [inline CRM/Upgrade/Incremental/php/FiveTwentyEight.php:86]: Shouldn't that be `($contactTypes as $contactType)` ?  And also $contactType (no 's') in the following lines for id and label?

Comment (seamuslee001) [reply CRM/Upgrade/Incremental/php/FiveTwentyEight.php]: thanks Fixed now

File: CRM/Upgrade/Incremental/php/FiveTwentyEight.php [hunk 2]
```diff
@@ -69,4 +69,30 @@ public function setPostUpgradeMessage(&$postUpgradeMessage, $rev) {
...
+
+  public static function populateMissingContactTypeName() {
+    $contactTypes = \Civi\Api4\ContactType::get()
+      ->setCheckPermissions(FALSE)
+      ->execute();
+    foreach ($contactTypes as $contactType) {
+      if (empty($contactType['name'])) {
+        \Civi\Api4\ContactType::update()
+          ->addWhere('id', '=', $contactType['id'])
+          ->addValue('name', ucfirst(CRM_Utils_String::munge($contactTypes['label'])))
```
Comment (aydun) [inline CRM/Upgrade/Incremental/php/FiveTwentyEight.php:90]: `$contactType['label']` not `$contactTypes['label']`

**Comment by aydun** [comment]
Should we also update any blank 'is_reserved' and 'is_active' since those defaults were also set in #17548 ?

**Comment by agh1** [comment]
Hi all just a side note here: I worry that adding `[REF]` to a PR like this that contains a database upgrade can obscure its impact.  In preparing release notes, we threw this down with the Miscellany section, but it turns out to have prompted issues upgrading on some sites.  I just wonder whether on principle a database upgrade might be incompatible with calling it a mere refactor.

**Comment by eileenmcnaughton** [comment]
It was a follow up to a refactor but that is probably fair comment

**Pull request merged**